python/ray/autoscaler/batching_node_provider.py

```diff
@@ -160,6 +167,11 @@ def non_terminated_nodes(self, tag_filters: Dict[str, str]) -> List[str]:
             workers_to_delete=set(),  # No workers to delete yet
         )
         all_nodes = list(self.node_data_dict.keys())
+        for node_id in all_nodes:
+            replica_index = self.node_data_dict[node_id].replica_index
+            # Only add node to map if it belongs to a TPU podslice
+            if replica_index is not None:
```

Comment:
Is it still possible for replica_index to be None?

Reply:
It's possible for replica_index to be None here, it's only guaranteed to not be None when returned by node_tags.